aea/protocols/generator/validate.py

```diff
@@ -823,6 +871,7 @@ def _validate_dialogue_section(
         ) = _validate_termination(
             cast(List[str], protocol_specification.dialogue_config["termination"]),
             performatives_set,
+            cast(Set[str], terminal_performatives_from_reply),
```

Comment:
Don't cast here; just treat optional type properly in the method...

Reply:
We know it won't be Optional (None). If it was, the previous `# Validate reply` block would return False and we won't reach this point. 

I moved cast at the beginning of the block for more clarity.